### DISCUSSION TITLE
Struggling to import

### DISCUSSION THREAD

**Discussion by NatSavBS** [pdfme/pdfme#130] [Q&A] [answered]
Hi All, 

Im struggling to import objects from @pdfme/generator (for use on node 16). all my tooling and intelisense is working with it however when i go to compile and run my typescript, i get.
```shell
import { BLANK_PDF, generate } from "@pdfme/generator";
         ^^^^^^^^^
SyntaxError: Named export 'BLANK_PDF' not found. The requested module '@pdfme/generator' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from '@pdfme/generator';
const { BLANK_PDF, generate } = pkg;
```
Pdfme comes in buth commonjs and esm so i dont know why its trying to pull in the commonjs version.
When i change the import to the type reccomended by the error, i get another error.
```shell
C:\...\project\node_modules\@pdfme\generator\dist\esm\src\index.js:1
import generate from './generate.js';
^^^^^^

SyntaxError: Cannot use import statement outside a module
```

I have attached my tsconfig, package.json and the file where I attempt to import pdfme
[files.zip](https://github.com/pdfme/pdfme/files/10433054/files.zip)

Thanks for any ideas/help!

**Comment by hand-dot** [comment, marked as answer]
Hi @NatSavBS 

Change module to CommonJS in `tsconfig.json`
```
"module": "CommonJS",   
```

Remove type =module in `package.json`
```
  "type": "module",
```

after that, you can compile
```
$ npx tsc pdfGenerator.ts
```

**Reply by NatSavBS** [reply]
hi @hand-dot 

That fixed it for me, i had to fiddle some of my other code that relied on esm module support but its all working now!
